tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsContextPropagatorTests.cs

```diff
@@ -4,40 +4,191 @@
+            {
+                Environment.SetEnvironmentVariable("DD_DATA_STREAMS_LEGACY_HEADERS", null);
+            }
+        }
+
+        [Fact]
+        public void Extract_WhenBothHeadersPresent_PrefersBase64Header()
+        {
+            var headers = new TestHeadersCollection();
+
+            var base64Context = new PathwayContext(
+                new PathwayHash(1234),
+                DateTimeOffset.UtcNow.AddSeconds(-5).ToUnixTimeNanoseconds(),
+                DateTimeOffset.UtcNow.ToUnixTimeNanoseconds());
+
+            var binaryContext = new PathwayContext(
+                new PathwayHash(5678),
+                DateTimeOffset.UtcNow.AddSeconds(-10).ToUnixTimeNanoseconds(),
+                DateTimeOffset.UtcNow.AddSeconds(-5).ToUnixTimeNanoseconds());
+
+            var encodedBase64ContextBytes = PathwayContextEncoder.Encode(base64Context);
+            var base64EncodedContext = Convert.ToBase64String(encodedBase64ContextBytes);
+            headers.Add(DataStreamsPropagationHeaders.PropagationKeyBase64, Encoding.UTF8.GetBytes(base64EncodedContext));
+
+            var encodedBinaryContextBytes = PathwayContextEncoder.Encode(binaryContext);
+            headers.Add(DataStreamsPropagationHeaders.PropagationKey, encodedBinaryContextBytes);
+
+            var extractedContext = DataStreamsContextPropagator.Instance.Extract(headers);
+
+            extractedContext.Should().NotBeNull();
+            extractedContext.Value.Hash.Value.Should().Be(base64Context.Hash.Value);
 
-    private static DateTimeOffset FromUnixTimeNanoseconds(long nanoseconds)
-        => DateTimeOffset.FromUnixTimeMilliseconds(nanoseconds / 1_000_000);
+            extractedContext.Value.Hash.Value.Should().NotBe(binaryContext.Hash.Value);
+            extractedContext.Value.PathwayStart.Should().NotBe(binaryContext.PathwayStart);
+            extractedContext.Value.EdgeStart.Should().NotBe(binaryContext.EdgeStart);
+        }
+
+        [Fact]
+        public void InjectedHeaders_HaveCorrectFormat()
+        {
+            var headers = new TestHeadersCollection();
+            var context = new PathwayContext(
+                new PathwayHash(0x12345678),
+                0x1122334455667788,
+                unchecked((long)0x99AABBCCDDEEFF00));
+
+            DataStreamsContextPropagator.Instance.Inject(context, headers);
+
+            headers.Values.Should().ContainKey(DataStreamsPropagationHeaders.PropagationKeyBase64);
+            var base64HeaderValueBytes = headers.Values[DataStreamsPropagationHeaders.PropagationKeyBase64];
+            var base64HeaderValue = Encoding.UTF8.GetString(base64HeaderValueBytes);
+            Assert.True(IsBase64String(base64HeaderValue), "Base64 header is not a valid Base64 string.");
+        }
+
+        [Fact]
+        public void InjectHeaders_WhenLegacyHeadersDisabled_DoesNotIncludeLegacyHeader()
+        {
+            Environment.SetEnvironmentVariable("DD_DATA_STREAMS_LEGACY_HEADERS", "false");
```

Comment:
Unfortunately, this is not going to work 😢 You'll need to extra the `Tracer.Instance` out of that method, so that you can create an instance here and pass it in to the tests. Otherwise this will break.